pkg/virt-controller/watch/migration/migration.go

```diff
@@ -951,20 +1024,78 @@ func (c *Controller) handlePreHandoffMigrationCancel(migration *virtv1.VirtualMa
 	return nil
 }
 
+func (c *Controller) getNodeSelectorsFromNodeName(nodeName string) (map[string]string, error) {
+	obj, exists, err := c.nodeStore.GetByKey(nodeName)
+	if err != nil {
+		return nil, err
+	}
+	res := make(map[string]string)
+	if exists {
+		node := obj.(*k8sv1.Node)
+		for key, value := range node.Labels {
+			if strings.HasPrefix(key, virtv1.HostModelCPULabel) || strings.HasPrefix(key, virtv1.HostModelRequiredFeaturesLabel) {
+				res[key] = value
+			}
+		}
+	}
+	return res, nil
+}
+
+func (c *Controller) updateVMIMigrationSourceWithPodInfo(migration *virtv1.VirtualMachineInstanceMigration, vmi *virtv1.VirtualMachineInstance) error {
+	vmiCopy := vmi.DeepCopy()
+
+	if migration.IsDecentralized() {
```

Comment:
```suggestion
	if !migration.IsDecentralized() {
		return nil
	}
```
To avoid indent

Reply:
Also looks like a good candidate for decentralized.go

Reply:
Done